### PR TITLE
Update tf_subnets with cidr_block input

### PR DESCRIPTION
**Pull request by comeanother**
## What

* Add option to use user-defined CIDR block

## Why

* We need option to create new subnets without collision with existing subnet CIDRs

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,19 +29,21 @@ module "subnets" {
...
+| stage                        | ``             | Stage (e.g. `prod`, `dev`, `staging`)                                                            | Yes      |
+| name                         | ``             | Name  (e.g. `bastion` or `db`)                                                                   | Yes      |
+| region                       | ``             | AWS Region where module should operate (e.g. `us-east-1`)                                        | Yes      |
+| vpc_id                       | ``             | The VPC ID where subnets will be created (e.g. `vpc-aceb2723`)                                   | Yes      |
+| igw_id                       | ``             | The Internet Gateway ID public route table will point to (e.g. `igw-9c26a123`)                   | Yes      |
+| cidr_block                   | ``             | The base CIDR block which will be divided into subnet CIDR blocks (e.g. `10.0.0.0/16`)           | Yes      |
+| vpc_default_route_table_id   | ``             | The scheduling expression. (e.g. cron(0 20 * * ? *) or rate(5 minutes)                           | No       |
```
Comment (aknysh) [inline README.md]: Please change the description for `vpc_default_route_table_id` to something like this:
```
The default route table for public subnets. Provides access to the Internet. If not set here, will be created.
```

**Comment by comeanother** [comment]
`${data.aws_availability_zones.available.count}` constantly returns `1`

**Comment by comeanother** [comment]
`${data.aws_availability_zones.available.count}` constantly returns `1`

File: README.md (hunk 1 above)
Comment (comeanother) [inline README.md]: @aknysh Nice catch! Thanks!

**Comment by aknysh** [comment]
I think you should use
`${data.aws_availability_zones.available.names.count}`

**Comment by comeanother** [comment]
`${length(data.aws_availability_zones.available.names)}` gives proper value.

**Comment by comeanother** [comment]
@aknysh Terraform can't resolve `${data.aws_availability_zones.available.names.count}` but says nothing.

**Comment by comeanother** [comment]
Description:
The subnets won't collide because we're calculate number of maximum required `newbits` based on number of availability zones in a region.

We're using the following expression:
`AZ` - number of availability zones in a region
`AZ * 2` - because we want 2 subnets (public and private per AZ)
```
newbits = ceil(log(AZ * 2, 2))
```

File: outputs.tf [hunk 2]
```diff
@@ -5,3 +5,7 @@ output "public_subnet_ids" {
 output "private_subnet_ids" {
   value = ["${aws_subnet.private.*.id}"]
 }
+
+output "test" {
```
Comment (osterman) [inline outputs.tf]: Remove test variable

File: private.tf [hunk 3]
```diff
@@ -10,8 +10,16 @@ resource "aws_subnet" "private" {
 
   vpc_id            = "${data.aws_vpc.default.id}"
   availability_zone = "${element(var.availability_zones, count.index)}"
-  cidr_block        = "${cidrsubnet(data.aws_vpc.default.cidr_block, length(var.availability_zones), length(var.availability_zones)  + count.index)}"
-  tags              = "${module.private_label.tags}"
+
+  cidr_block = "${
+    cidrsubnet(
```
Comment (osterman) [inline private.tf]: Next time, use a `null_resource` to define a trigger value for this complex calculation that is reused once again in `public.tf`

https://github.com/cloudposse/tf_label/blob/master/main.tf#L3

**Comment by osterman** [comment]
@comeanother also, it would be nice to add your explanation to the `README.md` about why you're using the log2 and ceil to compute the number of bits needed.

**Pull request merged**